src/plugins/matrixops/extract_shape.cpp

```diff
@@ -160,7 +197,7 @@ namespace phylanx { namespace execution_tree { namespace primitives
             index += 2;
 
         std::int64_t size = 0;
-        if (arg.has_annotation())
+        if (mode_ == dist_mode)
```

Comment:
We should probably still make sure the object actually has an annotation, otherwise we should fall back to the local case.